Binary/Loader/FileSystemLoader.php

```diff
@@ -30,78 +32,37 @@ class FileSystemLoader implements LoaderInterface
     protected $extensionGuesser;
 
     /**
-     * @var string
+     * @var LocatorInterface
      */
-    protected $dataRoots;
+    protected $locator;
 
     /**
-     * @param MimeTypeGuesserInterface  $mimeTypeGuesser
+     * @param MimeTypeGuesserInterface  $mimeGuesser
      * @param ExtensionGuesserInterface $extensionGuesser
      * @param string[]                  $dataRoots
+     * @param bool                      $secureLocator
      */
-    public function __construct(
-        MimeTypeGuesserInterface $mimeTypeGuesser,
-        ExtensionGuesserInterface $extensionGuesser,
-        $dataRoots
-    ) {
-        $this->mimeTypeGuesser = $mimeTypeGuesser;
+    public function __construct(MimeTypeGuesserInterface $mimeGuesser, ExtensionGuesserInterface $extensionGuesser, $dataRoots, $secureLocator = true)
```

Comment:
Yeah, fair enough. I hate multi-line constructors (unless it's well past 120 characters). That said, I know many are big on it (and an 80 character line limit). I'll revert. ;-) :+1: